### PR TITLE
Adds a method to handle out of bound UV in composer

### PR DESCRIPTION
**Pull request by stephomi**

### CODE REVIEW

File: sources/osgUtil/ComposerPostProcess.js [hunk 1]
```diff
@@ -118,6 +118,10 @@ var ComposerPostProcess = function() {
     // in case of dynamic resolution scaling : true will do a final upsampling
     // to full viewport size
     this._finalPassUpScaleToScreen = true;
+    // wrap uv stuffs
+    this._methodCWrapUV = 0;
```
Comment (Kuranes) [inline sources/osgUtil/ComposerPostProcess.js]: what the C stand for ? 
(not sure if color or something else?)

File: sources/osgUtil/ComposerPostProcess.js [hunk 2]
```diff
@@ -931,22 +935,61 @@ utils.createPrototypeObject(
...
-        _extractTextures: function(file) {
-            var infos = {
+        setMethodWrapUV: function(method, threshold) {
+            // hook every texture call to handle viewport/texture ratio
+            // when reading out of texture assigned zone
+
+            // 0 - do nothing
+            // 1 - black color when fetching out of bound
+            // 2 - clamp uv to thresold (usually vec2(1.0))
+            if (method !== undefined) this._methodWrapUV = method;
```
Comment (Kuranes) [inline sources/osgUtil/ComposerPostProcess.js]: this._methodCWrapUV ?

File: sources/osgUtil/ComposerPostProcess.js [hunk 3]
```diff
@@ -931,22 +935,61 @@ utils.createPrototypeObject(
...
+            // see setMethodWrapUV
+            var val = this._thresholdWrapUV.toExponential();
+            if (this._methodWrapUV === 1) {
+                var prefix = 'step((uv).x, ' + val + ') * step((uv).y, ' + val + ') * ';
+                bodySimple = prefix + bodySimple;
+                bodyNearest = prefix + bodyNearest;
+                bodyBias = prefix + bodyBias;
+            } else if (this._methodWrapUV === 2) {
+                var replaceValue = 'min(uv, vec2(' + val + '))';
+                bodySimple = bodySimple.replace('(uv)', replaceValue);
```
Comment (Kuranes) [inline sources/osgUtil/ComposerPostProcess.js]: strange was sure "replace" call needed a parameter for "global" to be able to replace multiple occurrences ?
```bodySimple.replace(/\(uv\)/g, replaceValue)```

File: sources/osgUtil/ComposerPostProcess.js (hunk 1 above)
Comment (stephomi) [reply sources/osgUtil/ComposerPostProcess.js]: typo

File: sources/osgUtil/ComposerPostProcess.js (hunk 3 above)
Comment (stephomi) [reply sources/osgUtil/ComposerPostProcess.js]: There's only 1 occurence of `(uv)` so it's fine.
But I can /g to be safer

**Pull request merged**